### ISSUE TITLE
🚨 [BUG] - --no-cleanup doesn't prevent execution of the cleanup block

### ISSUE THREAD

**Issue by VVX7** [facebookincubator/TTPForge#259]
### What happened?

Steps to repro:
1. Run a TTP with the --no-cleanup switch
2. Ignoring the cleanup block is not respected.
3. Run a TTP with -a cleanup=false param and ignoring cleanup block is respected.

### Relevant log output

```shell
# --no-cleanup
bash-3.2$ ./ttpforge run forgearmory//persistence/ssh/backdoor-ssh-authorized-keys/backdoor-ssh-authorized-keys.yaml -a rogue_key="aaaa" --no-cleanup
2023-08-19T23:58:39.270-0400	INFO	cmd/root.go:101	Using config file: /Users/vvx7/.ttpforge/config.yaml
2023-08-19T23:58:39.272-0400	INFO	blocks/ttps.go:199	[*] Validating Steps
2023-08-19T23:58:39.272-0400	INFO	blocks/ttps.go:210	[+] Finished validating steps
2023-08-19T23:58:39.272-0400	INFO	blocks/ttps.go:215	[+] Running current TTP: backdoor-authorized-keys
2023-08-19T23:58:39.272-0400	INFO	blocks/ttps.go:222	[+] Running current step: ensure-authorized_keys-present
2023-08-19T23:58:39.272-0400	INFO	blocks/basicstep.go:195	========= Executing ==========
2023-08-19T23:58:39.277-0400	INFO	blocks/basicstep.go:206	========= Done ==========
2023-08-19T23:58:39.277-0400	INFO	blocks/ttps.go:233	[+] Finished running step: ensure-authorized_keys-present
2023-08-19T23:58:39.277-0400	INFO	blocks/ttps.go:222	[+] Running current step: backup-authorized_keys
2023-08-19T23:58:39.277-0400	INFO	blocks/basicstep.go:195	========= Executing ==========
2023-08-19T23:58:39.284-0400	INFO	blocks/basicstep.go:206	========= Done ==========
2023-08-19T23:58:39.284-0400	INFO	blocks/ttps.go:233	[+] Finished running step: backup-authorized_keys
2023-08-19T23:58:39.284-0400	INFO	blocks/ttps.go:222	[+] Running current step: modify-authorized_keys
2023-08-19T23:58:39.284-0400	INFO	blocks/basicstep.go:195	========= Executing ==========
2023-08-19T23:58:39.289-0400	INFO	blocks/basicstep.go:206	========= Done ==========
2023-08-19T23:58:39.289-0400	INFO	blocks/ttps.go:233	[+] Finished running step: modify-authorized_keys
2023-08-19T23:58:39.289-0400	INFO	blocks/ttps.go:222	[+] Running current step: cleanup
2023-08-19T23:58:39.289-0400	INFO	blocks/basicstep.go:195	========= Executing ==========
2023-08-19T23:58:39.296-0400	INFO	blocks/basicstep.go:206	========= Done ==========
2023-08-19T23:58:39.296-0400	INFO	blocks/ttps.go:233	[+] Finished running step: cleanup
2023-08-19T23:58:39.296-0400	INFO	blocks/ttps.go:266	[*] Completed TTP
2023-08-19T23:58:39.296-0400	INFO	blocks/ttps.go:281	[*] No Cleanup Steps Found
bash-3.2$ 
bash-3.2$ cat ~/.ssh/authorized_keys
bash-3.2$ 
bash-3.2$ 


```bash
# -a cleanup=false
bash-3.2$ ./ttpforge run forgearmory//persistence/ssh/backdoor-ssh-authorized-keys/backdoor-ssh-authorized-keys.yaml -a rogue_key="aaaa" -a cleanup=false
2023-08-19T23:54:54.900-0400	INFO	cmd/root.go:101	Using config file: /Users/vvx7/.ttpforge/config.yaml
2023-08-19T23:54:54.901-0400	INFO	blocks/ttps.go:199	[*] Validating Steps
2023-08-19T23:54:54.901-0400	INFO	blocks/ttps.go:210	[+] Finished validating steps
2023-08-19T23:54:54.901-0400	INFO	blocks/ttps.go:215	[+] Running current TTP: backdoor-authorized-keys
2023-08-19T23:54:54.901-0400	INFO	blocks/ttps.go:222	[+] Running current step: ensure-authorized_keys-present
2023-08-19T23:54:54.901-0400	INFO	blocks/basicstep.go:195	========= Executing ==========
2023-08-19T23:54:54.906-0400	INFO	blocks/basicstep.go:206	========= Done ==========
2023-08-19T23:54:54.906-0400	INFO	blocks/ttps.go:233	[+] Finished running step: ensure-authorized_keys-present
2023-08-19T23:54:54.906-0400	INFO	blocks/ttps.go:222	[+] Running current step: backup-authorized_keys
2023-08-19T23:54:54.906-0400	INFO	blocks/basicstep.go:195	========= Executing ==========
2023-08-19T23:54:54.913-0400	INFO	blocks/basicstep.go:206	========= Done ==========
2023-08-19T23:54:54.913-0400	INFO	blocks/ttps.go:233	[+] Finished running step: backup-authorized_keys
2023-08-19T23:54:54.913-0400	INFO	blocks/ttps.go:222	[+] Running current step: modify-authorized_keys
2023-08-19T23:54:54.913-0400	INFO	blocks/basicstep.go:195	========= Executing ==========
2023-08-19T23:54:54.918-0400	INFO	blocks/basicstep.go:206	========= Done ==========
2023-08-19T23:54:54.918-0400	INFO	blocks/ttps.go:233	[+] Finished running step: modify-authorized_keys
2023-08-19T23:54:54.918-0400	INFO	blocks/ttps.go:222	[+] Running current step: cleanup
2023-08-19T23:54:54.918-0400	INFO	blocks/basicstep.go:195	========= Executing ==========
2023-08-19T23:54:54.922-0400	INFO	blocks/basicstep.go:206	========= Done ==========
2023-08-19T23:54:54.922-0400	INFO	blocks/ttps.go:233	[+] Finished running step: cleanup
2023-08-19T23:54:54.922-0400	INFO	blocks/ttps.go:266	[*] Completed TTP
2023-08-19T23:54:54.922-0400	INFO	blocks/ttps.go:281	[*] No Cleanup Steps Found
bash-3.2$ 
bash-3.2$ cat ~/.ssh/authorized_keys 
aaaa
```
```


### Details about your environment

### Details about your environment

System Version: macOS 13.5 (22G74)
System Arch: m1
Go: go version go1.21.0 darwin/amd64
TTPForge: v1.0.3

**Comment by l50** [comment]
Thanks for the bug, we'll take a look and fix!

**Comment by l50** [comment]
Sorry this took a minute. Your syntax wasn't quite right, here's how you'd run that TTP and opt out of the cleanup work:

```bash
ttpforge run forgearmory//persistence/ssh/backdoor-ssh-authorized-keys/backdoor-ssh-authorized-keys.yaml -a rogue_key="aaaa" --no-cleanup
```

Closing.

**Comment by l50** [comment]
This should be addressed with the merge of #342, thank you so much again for calling this to our attention @VVX7!
